src/structures/role.ts

```diff
@@ -2,10 +2,19 @@ import { Unpromise } from "../types/misc.ts";
 import { RoleData } from "../types/role.ts";
 
 export async function createRole(data: RoleData) {
+  const { tags, ...rest } = data;
+
+  const roleTags = {
+    botID: tags?.bot_id,
+    premiumSubscriber: Boolean(data.tags?.premium_subscriber),
```

Comment:
I think you were right before, this would fail if it is null. I didn't realize this is sent by discord as null that is very confusing sorry

Reply:
1 think is you can do tags instead of data.tags

Reply:
We should probably also add a comment above each one so it is documented in VSC

Reply:
JSDoc comments are added to the RoleData interface already.

Reply:
As for RoleTags interface, waiting for it to be documented; once it is, will do!